be/src/runtime/bufferpool/reservation_tracker.cc

```diff
@@ -281,14 +281,11 @@ bool ReservationTracker::TransferReservationTo(ReservationTracker* other, int64_
   // so this is all atomic.
   for (ReservationTracker* tracker : other_path_to_common) {
     tracker->UpdateReservation(bytes);
-    // We don't handle MemTrackers with limit in this function - this should always
-    // succeed.
-    DCHECK(tracker->mem_tracker_ == nullptr || !tracker->mem_tracker_->has_limit());
```

Comment:
I also found this bug when running in LSAN mode.
But the comment here is strange: `We don't handle MemTrackers with limit in this function`.
And if it has limit, the following `DCHECK(success)` should be removed too?
And what if it return false?

Reply:
We don't need to remove DCHECK(success).

I have removed `DCHECK(!_all_trackers[i]->has_limit());` in consume_local. So tracker->TryConsumeFromMemTracker will success.

Reply:
OK